core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableBatchNestedLoopJoin.java

```diff
@@ -124,8 +124,8 @@ public static EnumerableBatchNestedLoopJoin create(
       final RelMetadataQuery mq) {
     double rowCount = mq.getRowCount(this);
 
-    final double rightRowCount = right.estimateRowCount(mq);
```

Comment:
Can we please write a test for this? If it shouldn't be used then a test would be helpful to show why, and how this commit corrects that.

Reply:
+1